src/parquet/column/serialized-page-test.cc

```diff
@@ -0,0 +1,97 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include <gtest/gtest.h>
+
+#include "parquet/thrift/parquet_types.h"
+#include "parquet/thrift/util.h"
+#include "parquet/column/serialized-page.h"
+
+using std::string;
+
+namespace parquet_cpp {
+
+class TestSerializedPage : public ::testing::Test {
+ public:
+  void initSerializedPageReader(const uint8_t *buffer, size_t header_size,
+      parquet::CompressionCodec::type codec) {
+    std::unique_ptr<InputStream> stream;
+    stream.reset(new InMemoryInputStream(buffer, header_size));
+    page_reader_.reset(new SerializedPageReader(std::move(stream), codec));
+  }
+  void initStats(size_t stat_size, parquet::Statistics &stat) {
+    std::vector<char> stat_buffer;
+    stat_buffer.resize(stat_size);
+    for (int i = 0; i < stat_size; i++) {
+      (reinterpret_cast<uint8_t*>(stat_buffer.data()))[i] = i%255;
+    }
+    stat.__set_max(std::string(stat_buffer.data(), stat_size));
+  }
+  void initDataPage(parquet::DataPageHeader &data_page, parquet::Statistics &stat) {
```

Comment:
`parquet::DataPageHeader& data_page`. Same with all other usages of references and pointers in this patch